src/App.tsx

```diff
@@ -1,26 +1,240 @@
+/* eslint-disable prettier/prettier */
 /* eslint-disable max-len */
+/* eslint-disable jsx-a11y/label-has-associated-control */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useEffect, useRef, useState } from 'react';
 import { UserWarning } from './UserWarning';
-
-const USER_ID = 0;
+import * as todosService from './api/todos';
+import { Todo } from './types/Todo';
+import classNames from 'classnames';
+import { Header } from './components/Header';
+import { TodoList } from './components/TodoList';
+import { Footer } from './components/Footer';
+import { TodoFilter } from './types/FilterEnum';
 
 export const App: React.FC = () => {
-  if (!USER_ID) {
+  const [creatNewTodos, setCreateNewTodos] = useState('');
+  const [todoItem, setTodoItem] = useState<Todo[]>([]);
+  const [errorState, setStateError] = useState('');
+  const userId = todosService.USER_ID;
+  const [controlChecked, setControlChecked] = useState<number[]>([]);
+  const [filter, setFilter] = useState(TodoFilter.All);
+  const inputRef = useRef<HTMLInputElement>(null);
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  const [loadingNewItem, setLoadingNewItem] = useState(false);
+  const [arrTodos, setArrTodos] = useState<number[]>([]);
+  const [delLoader, setDelLoader] = useState<number | null>(null);
+  const [activeTodosCount, setActiveTodosCount] = useState(0);
+
+  useEffect(() => {
+    todosService
+      .getTodos()
+      .then(setTodoItem)
+      .catch(() => setStateError('Unable to load todos'));
+  }, []);
+
+  useEffect(() => {
+    if (errorState) {
+      const timer = setTimeout(() => {
+        setStateError(''); // Clear the error message after 3 seconds
+      }, 3000);
+
+      return () => clearTimeout(timer); // Clean up timer if component unmounts
+    }
+  }, [errorState]);
+
+
+  const getFilteredTodos = () => {
+    if (filter === TodoFilter.Active) {
+      return todoItem.filter(todo => !todo.completed);
+    }
+
+    if (filter === TodoFilter.Completed) {
+      return todoItem.filter(todo => todo.completed);
+    }
+
+    return todoItem;
+  };
+
+  const filteredTodos = getFilteredTodos();
+
+  useEffect(() => {
+    const activeCount = todoItem.filter(todo => !todo.completed).length;
+
+    setActiveTodosCount(activeCount);
+  }, [todoItem]); // Слідкуємо за змінами в todoItem
```

Comment:
Try to avoid comments. Good practice to write understandable code which will be "self commented".
If you need comments - good practice to write them in english.